src/phases/learn-move-phase.ts

```diff
@@ -12,11 +12,13 @@ import { PlayerPartyMemberPokemonPhase } from "./player-party-member-pokemon-pha
 
 export class LearnMovePhase extends PlayerPartyMemberPokemonPhase {
   private moveId: Moves;
+  private fromTM: boolean;
 
-  constructor(scene: BattleScene, partyMemberIndex: integer, moveId: Moves) {
+  constructor(scene: BattleScene, partyMemberIndex: integer, moveId: Moves, fromTM?: boolean) {
     super(scene, partyMemberIndex);
 
     this.moveId = moveId;
+    this.fromTM = fromTM || false;
```

Comment:
```suggestion
    this.fromTM = fromTM ?? false;
```
NIT: Safer operator for this situation